ibaqpy/bin/peptides2protein.py

```
@@ -0,0 +1,308 @@
+    :param aa_seq: Protein sequences from multiple database.
+    :return: One list contains nonstandard amoni acids and one remain sequence.
+    """
+    standard_aa = "ARNDBCEQZGHILKMFPSTWYV"
+    nonstandard_aa_lst = [aa for aa in aa_seq if aa not in standard_aa]
+    considered_seq = "".join([aa for aa in aa_seq if aa in standard_aa])
+    return nonstandard_aa_lst, considered_seq
+
+def extract_fasta(fasta:str, enzyme:str, proteins:List ,min_aa:int, max_aa:int, tpa: bool):
+    mw_dict = dict()
+    fasta_proteins = list()
+    FASTAFile().load(fasta, fasta_proteins)
```

Ruff (0.8.0)

`FASTAFile` may be undefined, or defined from star imports

(F405)